### PR TITLE
Documentation update

### PR DESCRIPTION
**Pull request by roman-khimov**
This is a port of #956 and #979 to the master branch. Many of these interops will actually go away in 3.0, but I think it's still worth having this updated documentation as a base for future development and documentation.

### CODE REVIEW

File: pkg/interop/iterator/iterator.go [hunk 1]
```diff
@@ -18,6 +18,15 @@ func Create(items []interface{}) Iterator {
 	return Iterator{}
 }
 
+// Concat concatenates two given iterators returning one that will range on
+// a first and then continue with b. Iterator positions are not reset for a
+// and b, so if any of them was already advanced by Next the resulting
+// Iterator will point at this new position and never go back to previous
+// key-value pairs. This function uses `Neo.Iterator.Concat` syscall.
+func Concat(a, b Iterator) Iterator {
```
Comment (fyrchik) [inline pkg/interop/iterator/iterator.go:26]: Another thing that can be important, that `Key()` call on an concatenated iterator will return "internal key" of the one of concatenated iterators, so there can be 2 items with the same key.
https://github.com/neo-project/neo/blob/master-2.x/neo.UnitTests/UT_ConcatenatedIterator.cs#L54

**Comment by fyrchik** [comment]
I think 3f9fb37 can be replaced with the one from #979, where we change `GetUnspentCoints` signature.

**Comment by roman-khimov** [comment]
I added both of them, it was a bit more convenient (less conflicts to resolve) and it gives proper credit.

File: pkg/interop/iterator/iterator.go (hunk 1 above)
Comment (roman-khimov) [reply pkg/interop/iterator/iterator.go:28]: Added a note.

**Pull request merged**